core/src/main/scala/org/broadinstitute/dsp/workbench/welder/AzureStorageInterp.scala

```diff
@@ -61,9 +108,50 @@ class AzureStorageInterp(implicit logger: StructuredLogger[IO]) extends CloudSto
       workingDir: Path,
       pattern: Regex,
       traceId: TraceId
-  ): fs2.Stream[IO, AdaptedGcsMetadataCache] = ???
+  )(implicit ev: Ask[IO, TraceId]): fs2.Stream[IO, Option[AdaptedGcsMetadataCache]] =
+    for {
+      traceId <- Stream.eval(ev.ask)
+      azureContainer = cloudStorageDirectory.containerName.asAzureCloudContainer
+      blob <- azureStorageService.listObjects(azureContainer,
+        Some(new ListBlobsOptions()
+          .setMaxResultsPerPage(1000)
+          .setPrefix(
+            cloudStorageDirectory
+              .blobPath.map(_.asString).getOrElse(""))
+        )
+      )
+      r <- pattern.findFirstIn(blob.getName) match {
+        case Some(blobName) =>
+          for {
+            localPath <- Stream.eval(IO.fromEither(getLocalPath(localBaseDirectory, cloudStorageDirectory.blobPath, blob.getName)))
+            localAbsolutePath <- Stream.fromEither[IO](Either.catchNonFatal(workingDir.resolve(localPath.asPath)))
+
+            result <- localAbsolutePath.toFile.exists() match {
+              case true => Stream(None).covary[IO]
+              case false =>
+                Stream.eval(mkdirIfNotExist(localAbsolutePath.getParent)) >>
+                  gcsToLocalFile(localAbsolutePath, AzurePath(azureContainer, BlobName(blobName)), traceId)
```

Comment:
this is where this implementation differs slightly from the googleStorageInterp, here we call the interface method `gcsToLocalFile` that exists within the welder azureStorageinterp already instead of a built-in function on the cloud sdk blob object